### ISSUE TITLE
Server Side Rendering 

### ISSUE THREAD

**Issue by ScottBirgel** [TucsonReactJS/primer#6]
Hello, I know this isn't a topic for this example really but I am having trouble with my basic isomorphic app. I am fetching some data from a postgres database and trying render it server side by setting the component's state to the data fetched. The problem however, is that when the client re-renders that component it does not have the same prop to render with and fails to render the same content therefore I am not gaining the benefits of a true isomorphic app. How do I pass the data fetched from the server as a prop to the client? How does data fetching work in an isomorphic app?

**Comment by bringking** [comment]
Hey @ScottBirgel, not worries about the forum, don't mind attempting to answer this here. 
- So in this case you basically just need to serialize the "as-rendered" state of your entire react application and pass it down to the client using something like [express-state](https://github.com/yahoo/express-state). This is easier said than done, which is why there are many frameworks to help out (like fluxible, or marty, or flummox). Essentially, you would need a helper for each of your stores that serializes their current state.
- Once you have that available client side, when you do your initial client rendering of React, you can pass that serialized state to your top level component using **React.createElement(yourAppComponent, yourDeserializedProps, yourDeserializedState)**

*Once your app is mounted and running with the server passed state, it can take over and do everything from there.

Does that make some sense? It's hard to tell exactly without seeing you implementation, but that is it in a nutshell.

**Comment by ScottBirgel** [comment]
Yeah that makes sense thanks for the answer! ill have to check out express-state. Is the serialized state the key prop that is passed to react component that allows react to use its diff functions? Also do you prefer webpack over browserify? There are so many packages to choose from in the npm world. Also kinda off topic but do you use passport for auth in your apps? 

Also, in general when is it appropriate to use server side rendering?
And I have heard many mixed opinions about how search engine crawlers are smarter now because they check for changes to the DOM after running javascript. Should I use server side rendering simpley because of performance or will it actually have any effect on the app's SEO
